examples/default.yaml

```diff
@@ -171,6 +171,12 @@ containerd:
 #     cat <<EOF > ~/.vimrc
 #     set number
 #     EOF
+# # `boot` is executed as part of cloud-init-local.service's early boot process
+# # See cloud-init docs for more info https://cloudinit.readthedocs.io/en/latest/topics/examples.html#run-commands-on-first-boot
+# - mode: boot
+#   script: |
+#     #!/bin/bash
```

Comment:
Does this work to execute the commands using `bash`?

I have my doubts because in `bootcmd` you can mix strings and list forms, and the list forms are executed by `execve` whereas the strings are executed by `sh`. So I doubt any hash-bang line has any effect on subsequent commands. But I don't know, so could you please verify, and then add a corresponding comment?

Reply:
Yeah, doesn't look like it. I used this boot script:

```yaml
- mode: boot
  script: |
    #!/bin/bash
    cat /proc/$$/cmdline | tr '\000' ' ' > /etc/bootcmdline
```

And it confirms that the script is launched via `/bin/sh`:

```console
$ lima cat /etc/bootcmdline
/bin/sh /run/cloud-init/tmp/tmp98wefsz7.sh
```

The hash-bang line will only be used when the file is loaded via the kernel; when passed directly to `/bin/sh` like this, it is just a comment.

So the hash-bang line should be removed from the sample script and the docs above should explain that the script will be executed by `/bin/sh`, and that you can't specify a hash-bang line.